v2/internal/genericarmclient/poller.go

```diff
@@ -10,15 +10,16 @@ import (
 	"net/http"
 
 	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
-	armruntime "github.com/Azure/azure-sdk-for-go/sdk/azcore/arm/runtime"
+	azcoreruntime "github.com/Azure/azure-sdk-for-go/sdk/azcore/runtime"
 	"github.com/pkg/errors"
 )
 
 // PollerResponse is the response from issuing a PUT to Azure. It contains a poller (for polling the long-running
 // operation URL) and a RawResponse containing the raw HTTP response.
 type PollerResponse struct {
 	// Poller contains an initialized poller.
-	Poller *azcore.Poller
+	//TODO: Keeping the generic type as string here.. Should we narrow it down to a type?
+	Poller *azcoreruntime.Poller[string]
```

Comment:
I don't think we want this to be `string`. I based `generic_client.go` on https://github.com/Azure/azure-sdk-for-go/blob/main/sdk/resourcemanager/resources/armresources/zz_generated_client.go

If you look at that, the poller is using `GenericResource` https://github.com/Azure/azure-sdk-for-go/blob/09a0280a389ea06855ce9c7e78afd1fdb3bf2d99/sdk/resourcemanager/resources/armresources/zz_generated_models.go#L922 as its generic type. I had deleted the `GenericResource` because originally it was only used for the input type and we're instead using `interface{}`.

Possibly we should have a generic type we're deserializing responses into, as subset of `GenericResource` that looks something like:
```
type GenericResource struct {
	// The kind of the resource.
	Kind *string `json:"kind,omitempty"`

	// READ-ONLY; Resource ID
	ID *string `json:"id,omitempty"

	// READ-ONLY; Resource name
	Name *string `json:"name,omitempty"

	// READ-ONLY; Resource type
	Type *string `json:"type,omitempty"
}
```

then we could use that here?

Reply:
It doesn't really matter that much right now because nothing uses the well-typed response from the poller... but in the future it might be useful.